### ISSUE TITLE
Elasticsearch service install on windows does not respect shell variables 

### ISSUE THREAD

**Issue by jakommo** [elastic/elasticsearch#24829]
**Elasticsearch version**: 5.4.0

**Plugins installed**: []

**JVM version** (`java -version`): Java 8 u131

**OS version** (`uname -a` if on a Unix-like system): Windows 2012R2 (also tested on Windows 10)

**Description of the problem including expected versus actual behavior**:

The [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/5.4/windows.html#windows-service-settings) is saying:
> The Elasticsearch service can be configured prior to installation by setting the following environment variables (either using the set command from the command line, or through the System Properties->Environment Variables GUI).

But when the variables are set using the `set` command, they are not picked up by the service installer and use the defaults instead. When set via the GUI, it works as expected. 

**Steps to reproduce**:

1. Unzip elasticsearch-5.4.0.zip to C:\TEMP\ESServiceTestA
2. Open a command window with admin privileges and enter the following commands:
a. _cd C:\TEMP\ESServiceTestA\elasticsearch-5.4.0\bin_
b. set DATA_DIR=C:\TEMP\ESServiceTestA\data
c. set LOG_DIR=C:\TEMP\ESServiceTestA\logs
d. elasticsearch-service install ESServiceTestA
e. sc start ESServiceTestA

Result:
1. After executing the new service the following folders are created and being used. 
`C:\TEMP\ESServiceTestA\elasticsearch-5.4.0\data` and `C:\TEMP\ESServiceTestA\elasticsearch-5.4.0\logs` instead of the ones specified in `DATA_DIR` and `LOG_DIR`.

**Comment by ismael-hasan** [comment]
Using the service manager GUI, under the Java tab, it seems that the variables are picked, but not applied. The GUI shows them: 
```
-Des.default.path.logs=C:\temp\logs
-Des.default.path.data=C:\temp\data
```

Changing them to 
```
-Des.path.logs=C:\temp\logs
-Des.path.data=C:\temp\data
```

effectively sets both directories when starting the service. The same happens with `CONF_DIR`.

**Comment by Toca74** [comment]
Here's an alternative solution that avoids the need for making changes in the service manager GUI after installation:

> set ES_JAVA_OPTS=-Des.path.data=c:\temp\data -Des.path.logs=c:\temp\logs

...and then run the install script.

**Comment by olcbean** [comment]
hey @Mpdreamz @javanna 
This should be a fairly simple fix for 5.x and I would like to have a go at it. I figured I will ask before doing the work, as the issue does not have an `adoptme` tag.
Thanks!

**Comment by jasontedor** [comment]
Closed by #26875
